specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;
+                        MessageBoxImage icnMessageBox = MessageBoxImage.None;
+                        MessageBoxResult resultbox = MessageBox.Show(text, caption, btnMessageBox, icnMessageBox);
+                        switch (resultbox)
+                        {
+                            case MessageBoxResult.Yes:
+                                ProcessStartInfo info = new ProcessStartInfo
+                                {
+                                    FileName = args.Uri,
+                                    UseShellExecute = true
+                                };
+                                Process.Start(info);
+                                break;
+
+                            case MessageBoxResult.No:
+                                break;
+
+                            case MessageBoxResult.Cancel:
+                                break;
+                        }
+
+                    }
+                    
+                }
+            }, null);
+        }
+    };
+}
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+    
+```IDL
+// This is the ICoreWebView2_16 interface.
+[uuid(cc39bea3-d6d8-471b-919f-da253e2fbf03), object, pointer_default(unique)] 
+interface ICoreWebView2_16 : ICoreWebView2_15 {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// The `LaunchingExternalProtocol` event is raised when a launch request is made to
+  /// a protocol that is registered with the OS. 
+  /// The `LaunchingExternalProtocol` event may suppress the default dialog
+  /// or replace the default dialog with a custom dialog.
+  ///
+  /// If a deferral is not taken on the event args, the external protocol launch is 
+  /// blocked until the event handler returns.  If a deferral is taken, the
+  /// external protocol launch is blocked until the deferral is completed.
+  /// The host also has the option to cancel the protocol launch.
+  ///
+  /// The `NavigationStarting` and `NavigationCompleted` events will be raised,
+  /// regardless of whether the `Cancel` property is set to `TRUE` or
+  /// `FALSE`. The `SourceChanged`, `ContentLoading`, and `HistoryChanged` events 
+  /// will not be raised regardless of this property. 
+  /// The `LaunchingExternalProtocol` event will be raised after the 
+  /// `NavigationStarting` event and before the `NavigationCompleted` event.
+  /// The default settings will also be updated upon navigation to an external
+  /// protocol. 
+  ///
+  /// If the request is made from a trustworthy origin
+  /// (https://w3c.github.io/webappsec-secure-contexts#potentially-trustworthy-origin) 
+  /// a checkmark box will be displayed on the default browser UI that gives the user 

```

Comment:
Is this description of the checkmark box part of this Event?  Or is this a comment about Browser UI today?

Reply:
This comment is regarding the default browser UI.

Reply:
Is there a way for me to include this in my custom/replacement dialog? (A way to find out that it's a trustworthy origin?)

Reply:
We should make this text more generic and just say WebView2 may not display the dialog based on the user settings, browser settings, trusted origins (or whatever) but that the event will still be raised.